lms/djangoapps/courseware/views.py

```diff
@@ -402,6 +397,14 @@ def _index_bulk_op(request, course_key, chapter, section, position):
                 and user_must_complete_entrance_exam(request, user, course):
             log.info(u'User %d tried to view course %s without passing entrance exam', user.id, unicode(course.id))
             return redirect(reverse('courseware', args=[unicode(course.id)]))
+
+    # Gated Content Check
+    gated_content = gating_api.get_gated_content(course.id, user)
```

Comment:
This will need to be load-tested to make sure that it doesn't adversely impact LMS page loading time.

https://openedx.atlassian.net/wiki/display/EdxOps/How+to+Run+Performance+Tests